renku_notebooks/api/schemas.py

```diff
@@ -224,6 +224,14 @@ def resolve_gpu_fieldname(self, in_data, **kwargs):
         return in_data
 
 
+class ServerStatus(Schema):
+    state = fields.String(
+        required=True,
+        validate=lambda x: x in ["running", "starting", "stopping", "failed"],
```

Comment:
```suggestion
        validate=validate.OneOf(["running", "starting", "stopping", "failed"]),
```
I'm a big fan of using Enum for these kinds of things. Right now they're just disconnected strings in different parts of the file, but an Enum could make that safer and unify it. 
E.g.
```
class ServerStatusEnum(Enum):
    Running = "running"
    Starting = "starting"

    @classmethod
    def list(cls):
        return list(map(lambda c: c.value, cls))

[...]

validate=validate.OneOf(ServerStatusEnum.list())

[...]

return {
    "state": ServerStatusEnum.Running.value,
}
```

Reply:
This is really neat. Thanks I added it.